### PR TITLE
Add build variants for vtk 8.1.2 and vtk 9

### LINKED ISSUE #296: With vtk8 Tiff reading tests fail whereas with vtk9 they do not
In an environment with current master and vtk8.1.2 I get the following errors:

> ======================================================================
> FAIL: test_tiff_cropped_reader (test_cropped_readers.TestCroppedReaders)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "C:\Users\lhe97136\Work\CCPi\CILViewer\Wrappers\Python\test\test_cropped_readers.py", line 107, in test_tiff_cropped_reader
>     self.check_values(target_z_extent, reader.GetOutput())
>   File "C:\Users\lhe97136\Work\CCPi\CILViewer\Wrappers\Python\test\test_cropped_readers.py", line 66, in check_values
>     np.testing.assert_array_equal(cropped_array, read_cropped_array)
>   File "C:\Users\lhe97136\Miniconda3\envs\vtk8-viewer\lib\site-packages\numpy\testing\_private\utils.py", line 935, in assert_array_equal
>     verbose=verbose, header='Arrays are not equal')
>   File "C:\Users\lhe97136\Miniconda3\envs\vtk8-viewer\lib\site-packages\numpy\testing\_private\utils.py", line 844, in assert_array_compare
>     raise AssertionError(msg)
> AssertionError:
> Arrays are not equal
> 
> Mismatched elements: 72 / 72 (100%)
> Max absolute difference: 250
> Max relative difference: 6.94444444
>  x: array([[[24, 25, 26, 27, 28, 29],
>         [30, 31, 32, 33, 34, 35],
>         [36, 37, 38, 39, 40, 41],...
>  y: array([[[42, 43, 44, 45, 46, 47],
>         [36, 37, 38, 39, 40, 41],
>         [30, 31, 32, 33, 34, 35],...
> 
> ======================================================================
> FAIL: test_tiff_resample_reader (test_resample_readers.TestResampleReaders)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "C:\Users\lhe97136\Work\CCPi\CILViewer\Wrappers\Python\test\test_resample_readers.py", line 152, in test_tiff_resample_reader
>     self.resample_reader_test1(reader, 100 * 8 * 2)
>   File "C:\Users\lhe97136\Work\CCPi\CILViewer\Wrappers\Python\test\test_resample_readers.py", line 134, in resample_reader_test1
>     np.testing.assert_array_equal(np.asfortranarray(self.input_3D_array), resulting_array)
>   File "C:\Users\lhe97136\Miniconda3\envs\vtk8-viewer\lib\site-packages\numpy\testing\_private\utils.py", line 935, in assert_array_equal
>     verbose=verbose, header='Arrays are not equal')
>   File "C:\Users\lhe97136\Miniconda3\envs\vtk8-viewer\lib\site-packages\numpy\testing\_private\utils.py", line 844, in assert_array_compare
>     raise AssertionError(msg)
> AssertionError:
> Arrays are not equal
> 
> Mismatched elements: 300 / 300 (100%)
> Max absolute difference: 65530
> Max relative difference: 2184.33333333
>  x: array([[[  0,   1,   2,   3,   4,   5],
>         [  6,   7,   8,   9,  10,  11],
>         [ 12,  13,  14,  15,  16,  17],...
>  y: array([[[ 54,  55,  56,  57,  58,  59],
>         [ 48,  49,  50,  51,  52,  53],
>         [ 42,  43,  44,  45,  46,  47],...
> 
> ----------------------------------------------------------------------
> Ran 19 tests in 2.704s
> 
> FAILED (failures=2)

### PR DESCRIPTION
**Pull request by lauramurgatroyd**
This prevents us from needing to release 2 different versions of the viewer every time we make a release.

I tested building locally and making environments installing from local channel with a vtk9 or vtk 8.1.2 and they both seemed to be happy installing.

Also closes #296

### CODE REVIEW

**Comment by lauramurgatroyd** [comment]
The tests failing are test_tiff_resample_reader and the tiff cropped reader test. Not sure how them failing is related to this PR

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,16 @@ ccpi_viewer.egg-info
...
+.vscode/*
+
+# Image files:
+current_render*.png
+*.mha
+*.mhd
+*.raw
+*.h5
+*.npy
+*.nxs
```
Comment (lauramurgatroyd) [inline .gitignore:14]:
```suggestion
```

**Comment by lauramurgatroyd** [comment]
See #296 for details on failing tiff tests, should now be fixed.

**Comment by lauramurgatroyd** [comment]
Not sure why the action took 1h20 to run but it works now!!

**Comment by lauramurgatroyd** [comment]
It took 8 minutes to build on vishighmem....

**Comment by lauramurgatroyd** [comment]
Yay, it only takes 20 mins now! Not too bad as it used to take 10 mins to build + test one thing and now takes 20 mins to build + test 2 things

File: Wrappers/Python/ccpi/viewer/utils/conversion.py [hunk 2]
```diff
@@ -1189,7 +1191,13 @@ def RequestData(self, request, inInfo, outInfo):
                 reader.Modified()
                 reader.Update()
                 # print(reader.GetOutput().GetScalarComponentAsDouble(0, 0, 0, 0))
-                outData.ShallowCopy(reader.GetOutput())
+                data = reader.GetOutput()
+                if self.__class__.__name__ == "cilTIFFResampleReader" and VTK_MAJOR_VERSION == 8:
+                    # In this case we have to flip along the y axis
```
Comment (paskino) [inline Wrappers/Python/ccpi/viewer/utils/conversion.py:1196]: Why is that? Is there any VTK change that we can quote? Or is it just a workaround?

File: Wrappers/Python/ccpi/viewer/utils/conversion.py [hunk 3]
```diff
@@ -1232,7 +1240,7 @@ def RequestData(self, request, inInfo, outInfo):
                                            element_spacing[1] / xy_axes_magnification,
                                            element_spacing[2] / z_axis_magnification)
                 # resampled data
-                resampled_image = outData
+                resampled_image = vtk.vtkImageData()
```
Comment (paskino) [inline Wrappers/Python/ccpi/viewer/utils/conversion.py:1243]: Good spot, this was a serious bug

File: Wrappers/Python/ccpi/viewer/utils/conversion.py (hunk 2 above)
Comment (lauramurgatroyd) [reply Wrappers/Python/ccpi/viewer/utils/conversion.py:1196]: See https://github.com/vais-ral/CILViewer/issues/296#issuecomment-1199037554

Comment (lauramurgatroyd) [reply Wrappers/Python/ccpi/viewer/utils/conversion.py:1196]: Sorry, I posted the wrong link, meant to post this one https://github.com/vais-ral/CILViewer/issues/296#issuecomment-1198007102

File: Wrappers/Python/ccpi/viewer/utils/conversion.py [hunk 4]
```diff
@@ -1893,7 +1950,14 @@ def RequestData(self, request, inInfo, outInfo):
                 sa.InsertNextValue(el)
             reader.SetFileNames(sa)
             reader.Update()
-            outData.ShallowCopy(reader.GetOutput())
+            data = reader.GetOutput()
+            if VTK_MAJOR_VERSION == 8:
+                # In this case we have to flip along the y axis
+                data = Converter.vtk2numpy(reader.GetOutput())
+                data = np.flip(data, axis=1)
+                data = Converter.numpy2vtkImage(data)
```
Comment (paskino) [inline Wrappers/Python/ccpi/viewer/utils/conversion.py:1958]: This is not necessary now, right?

Comment (lauramurgatroyd) [reply Wrappers/Python/ccpi/viewer/utils/conversion.py:1958]: not sure how I missed this one

Comment (lauramurgatroyd) [reply Wrappers/Python/ccpi/viewer/utils/conversion.py:1958]: or how it passed the tests becuase VTK_MAJOR_VERSION doesn't exist.......

**Review by paskino — CHANGES_REQUESTED** [review]
I think you have a leftover check of VTK version

**Comment by lauramurgatroyd** [comment]
Thanks, I have deleted it now! Unit tests passed locally, but we should wait for them to pass here too

**Review by paskino — APPROVED** [review]

**Pull request merged**